### PR TITLE
Change hitTest signatures to be non-nullable

### LINKED ISSUE #87791: Rendering contains some unnecessary transform asserts
When `addWithPaintOffset` has the `offset` parameter, it automatically does what is asserted in places like this

https://github.com/flutter/flutter/blob/025397ae3f925b316991912977367d3263cc4432/packages/flutter/lib/src/widgets/single_child_scroll_view.dart#L665-L668

### PR DESCRIPTION
**Pull request by nt4f04uNd**
Fixes https://github.com/flutter/flutter/issues/87791

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/rendering/editable.dart [hunk 1]
```diff
@@ -3292,13 +3292,8 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
       final bool isHit = result.addWithPaintTransform(
         transform: transform,
         position: position,
-        hitTest: (BoxHitTestResult result, Offset? transformed) {
-          assert(() {
-            final Offset manualPosition = (position - textParentData.offset) / textParentData.scale!;
-            return (transformed!.dx - manualPosition.dx).abs() < precisionErrorTolerance
-              && (transformed.dy - manualPosition.dy).abs() < precisionErrorTolerance;
-          }());
```
Comment (nt4f04uNd) [inline packages/flutter/lib/src/rendering/editable.dart:3300]: This is the same as other asserts, but with more shenanigans around precision error tolerance and transform

**Review by goderbauer — APPROVED** [review]
LGTM, thnaks for the cleanup.

Will need a test or an exception from Hixie.

**Comment by Hixie** [comment]
test-exempt: code refactor with no effective semantic change

**Comment by goderbauer** [comment]
Sorry for the delay, could you rebase this with master to retrigger the google test?

**Comment by nt4f04uNd** [comment]
It looks like it doesn't want to run.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**